### PR TITLE
generate makefile

### PR DESCRIPTION
**Pull request by cnell-usgs**
generate targets makefile

### CODE REVIEW

**Comment by cnell-usgs** [comment]
image


This took me a minute to sort out. I kept getting an error running `targets::tar_make()` that said `dynamic files (targets with format = "file") must return character vectors of file or directory paths.` That forced me to add another function as a wrapper to `write_csv` to print the output location...this seems a bit weird and likely I am missing something.

**Comment by cnell-usgs** [comment]
Ah yes, now I see 
"There are two types of targets - objects and files. **If your target is a file, you need to add format = "file" to your tar_target() call and the command needs to return the filename of the new file.**"

File: _targets.R [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  # Create a plot
+  tar_target(
+    figure_1_png,
+    plot_data(plot_out = "3_visualize/out/figure_1.png", eval_data), 
+    format = "file"
+  ),
+  # Save the processed data
+  tar_target(
+    model_summary_results_csv,
+    write_rmse(eval_data, file_out = "2_process/out/model_summary_results.csv"),
```
Comment (lindsayplatt) [inline _targets.R:28]: I can't remember if the course has talked about this yet (so potentially _**spoiler alert**_) but one way to get around having to write a custom function to make sure that the target command returns the filepath for file targets is to use `{ }` and put your code directly in the targets makefile. Not something you want to do when there are lots of lines of code, but nice when it is something simple, such as needing to return the filepath.

E.g.
```r
tar_target(
  model_summary_results_csv,
  {
    file_out <- "2_process/out/model_summary_results.csv"
    write_csv(eval_data, file = file_out)
    return(file_out)
  }, 
  format = "file"
),
```

**Review by lindsayplatt — APPROVED** [review]
Nice! One small thing - you shouldn't need to commit things in the `/out/` folders. I feel like this is not super clear in the course instructions, so I am going to go add that as a "to-do".

**Comment by lindsayplatt** [comment]
Feel free to self-merge when you're ready!

**Pull request merged**